addons/middle-click-popup/userscript.js

```diff
@@ -375,8 +423,30 @@ export default async function ({ addon, msg, console }) {
       preventDefault: function () {},
       target: selectedPreview.svgBlock,
     };
-    if (workspace.getGesture(fakeEvent)) {
-      workspace.startDragWithFakeEvent(fakeEvent, newBlock);
+    if (Blockly.registry) {
+      // new Blockly expects a pointerdown event
+      fakeEvent.type = "pointerdown";
+      fakeEvent.pointerType = "mouse";
+      // If the block is being dragged using the mouse or touch, the correct pointerId
+      // needs to be set so that Blockly recognizes the associated pointerup event.
+      // If the block is selected using the keyboard, pointerId will be undefined and a
+      // click will be necessary to end the "drag".
```

Comment:
This never happens and I don't see any errors in the console.

Reply:
The comment could probably be phrased better - it's referring to using arrow keys and enter to select a block. In that case, e.pointerId will be undefined (because e is a keyboard event), which doesn't cause any errors but I wanted people reading the code to know it might happen.